python/src/plugins/headless/goat_plugins/crossmint_headless_checkout/service.py

```diff
@@ -0,0 +1,316 @@
+                    )
+                    
+                # Decode the serialized transaction
+                transaction_bytes = base58.b58decode(serialized_transaction)
+                
+                # Send the transaction using the wallet
+                result = await wallet_client.send_transaction(
+                    transaction_bytes=transaction_bytes
+                )
+                
+                return {"order": order, "txId": result["signature"]}
+                
+            # Handle EVM transactions
+            if self._is_evm_blockchain(payment_method):
+                # Check for EVM wallet client (including web3 version)
+                is_evm_wallet = isinstance(wallet_client, EVMWalletClient)
+                if Web3EVMWalletClient:
+                    is_evm_wallet = is_evm_wallet or isinstance(wallet_client, Web3EVMWalletClient)
+                
+                if not is_evm_wallet:
+                    logger.error(f"Wallet client type: {type(wallet_client)}")
+                    logger.error(f"Expected EVMWalletClient or Web3EVMWalletClient")
+                    raise Exception(
+                        "EVM wallet client required. Use an evm wallet client, or change the payment method to one supported by your wallet client"
+                    )
+                    
+                # Parse the serialized transaction to extract to, value, and data
+                if not serialized_transaction.startswith("0x"):
+                    serialized_transaction = f"0x{serialized_transaction}"
+                    
+                logger.info(f"Paying order: {order.get('orderId')}")
+                logger.info(f"Parsing serialized transaction: {serialized_transaction[:50]}...")
+                
+                try:
+                    # Use web3.py to decode the transaction
+                    from web3 import Web3
+                    from eth_utils import to_bytes, to_hex
+                    import rlp
+                    
+                    # Remove '0x' prefix and convert to bytes
+                    tx_bytes = to_bytes(hexstr=serialized_transaction)
+                    
+                    # Simple approach: try RLP decoding first (works for both legacy and most typed transactions)
+                    try:
+                        decoded = rlp.decode(tx_bytes)
+                        # Legacy transaction format: [nonce, gasPrice, gasLimit, to, value, data, v, r, s]
+                        transaction_data = {
+                            "to": to_hex(decoded[3]) if decoded[3] else None,
+                            "value": int.from_bytes(decoded[4], 'big') if decoded[4] else 0,
+                            "data": to_hex(decoded[5]) if decoded[5] else "0x",
+                        }
+                    except Exception:
+                        # If RLP fails, try to handle as typed transaction by removing type byte
+                        if tx_bytes[0] <= 0x7f:
```

Comment:
hmm interesting.. not sure we need this for the txs we will be dealing with in this API, id remove for now

Reply:
Meaning handling this whole exception